app/services/katello/content_unit_indexer.rb

```diff
@@ -38,6 +38,12 @@ def import_all(filtered_indexing = false)
             end
           end
 
+          # Even after this bug (https://github.com/pulp/pulp_rpm/issues/2821) is fixed,
+          # it is possible to have duplicate errata asosociated to a repo.
+          if @content_type.label == 'erratum'
```

Comment:
The units in `to_insert` and `@final_values` below both seem to come from `units_from_pulp`.  Here, `to_insert` comes from the iteration through `units_from_pulp`. Below, `@final_values` comes from `@values` through `::Katello::ContentUnitIndexer.pulp_id_to_id_map`.  `@values` is populated through `def push`, which is also fed by `units_from_pulp`.

With all this in mind, would it make sense to filter out the duplicate errata in `units_from_pulp` itself so it doesn't need to be done twice?

Reply:
Likewise, I noticed there's a `remove_duplicate` method being called -- would it make sense for this logic to live there?

Reply:
Although, this duplicate checking should likely happen before the content units are used, so `remove_duplicates` might be too late. Perhaps it could happen in `units_from_pulp` itself.

Reply:
I'm curious why you needed to remove duplicates in two places rather than one.

Reply:
Now this is the only one, nice :)